google-cloud-firestore/src/main/java/com/google/cloud/firestore/UpdateBuilder.java

```diff
@@ -47,17 +49,47 @@
  * Abstract class that collects and bundles all write operations for {@link Transaction} and {@link
  * WriteBatch}.
  */
-public abstract class UpdateBuilder<T extends UpdateBuilder> {
+public abstract class UpdateBuilder<T> {
```

Comment:
This implementation is now very different from Node. Do you think we can easily get Node to match? I am a little hesitant to provide two drastically different reference implementations.

Reply:
Let's live as is. Java remains outlier. We need to point that out in our docs for the contractors.